### PR TITLE
Update pattern

### PR DESCRIPTION
**Pull request by Philmod**

### CODE REVIEW

File: dist/react-maskedinput.js [hunk 1]
```diff
@@ -42,6 +42,25 @@ var MaskedInput = React.createClass({displayName: "MaskedInput",
...
+
+  componentDidUpdate:function(prevProps) {
+    if (prevProps.pattern !== this.props.pattern && this.mask.selection.start) {
+      this._updateInputSelection()
+    }
+  },
+
+  _updatePattern: function(props) {
+    this.mask.setPattern(props.pattern, {
+      value: this.props.value,
```
Comment (muffinresearch) [inline dist/react-maskedinput.js]: I think this should use `this.mask.getRawValue()` - this way it's not going to put placeholders where chars should be when the pattern changes.

**Comment by muffinresearch** [comment]
This looks to address most of the issues I noted here: https://github.com/insin/react-maskedinput/issues/8#issuecomment-125251570

Perhaps if this could get rebased against master (squashing the commits) this could be a contender for being really close for a fix to #8 \o/

See https://github.com/insin/react-maskedinput/issues/8#issuecomment-125538427 for the one remaining nit. Going from a short pattern to a longer pattern via paste, the selection is not at the the end of the string. We can probably look at fixing that separately in any case.

**Comment by Philmod** [comment]
@muffinresearch You got it.

**Comment by muffinresearch** [comment]
@insin would you have some time to take a look at this patch in relation to #8 ?

**Comment by muffinresearch** [comment]
Hmm not sure what's happening but the dist file looks to be out of date e.g. it's missing the `componentWillReceiveProps` that's in the .jsx flle. Also testing the changes I'm no longer seeing pattern updates rendering. Can you see if still works for you?

**Comment by SimeonC** [comment]
Hey, just grabbed this one to see if it fixed my issue - PR has issues with pattern vs match so the following code snippet is the fixed version of the code (No time to make PR sorry). AFAIK this works for all issues in #8 as well (didn't extensively test though).

``` js
componentWillUpdate(nextProps, nextState) {
    if (nextProps.mask !== this.props.mask) {
      this._updatePattern(nextProps)
    }
  },

  componentDidUpdate(prevProps) {
    if (prevProps.mask !== this.props.mask && this.mask.selection.start) {
      this._updateInputSelection()
    }
  },

  _updatePattern: function(props) {
    this.mask.setPattern(props.mask, {
      value: this.mask.getRawValue(),
      selection: getSelection(this.input)
    });
  },
```

**Comment by iamdustan** [comment]
Closed by #53

**Pull request closed without merging**